src/base/utils/gecko.jl

```diff
@@ -70,17 +70,30 @@ _gecko_gene_product_coupling(model::GeckoModel) =
 Compute the part of the coupling for [`GeckoModel`](@ref) that limits the total
 mass of each group of gene products.
 """
-_gecko_mass_group_coupling(model::GeckoModel) =
-    let
-        tmp = [
-            (row, i, val) for (i, col) in enumerate(model.columns) for
-            (row, val) in col.mass_group_coupling
-        ]
-        sparse(
-            [row for (row, _, _) in tmp],
-            [col for (_, col, _) in tmp],
-            [val for (_, _, val) in tmp],
-            length(model.coupling_row_mass_group),
-            length(model.columns),
-        )
-    end
+function _gecko_mass_group_coupling(model::GeckoModel)
+    tmp = [ # mm = molar mass, mg = mass group, i = row idx, j = col idx
+        (i, j, mm) for (i, mg) in enumerate(model.coupling_row_mass_group) for
+        (j, mm) in zip(mg[2], mg[3])
+    ]
+    sparse(
+        [i for (i, _, _) in tmp],
+        [j for (_, j, _) in tmp],
+        [mm for (_, _, mm) in tmp],
+        length(model.coupling_row_mass_group),
+        n_genes(model),
+    )
+end
+
+"""
+    flux_dict(model::GeckoModel, opt_model)
+
+Returns the fluxes (not gene product concentrations) of the model as a
+reaction-keyed dictionary, if solved. See [`protein_dict`](@ref) for a 
+function to get the gene product concentrations.
+"""
+flux_dict(model::GeckoModel, opt_model) =
+    is_solved(opt_model) ?
+    Dict(
+        fluxes(model) .=>
+            reaction_flux(model)' * value.(opt_model[:x])[1:n_reactions(model)],
+    ) : nothing
```

Comment:
this _must_ be derivable from `reaction_fluxes` and `fluxes`, having a custom magic overload for just GeckoModel is confusing. Do the `reaction_fluxes` right instead.

Reply:
(After the quick discussion: make a separate `enzyme_dict`, anyone can mix the dictionaries from `flux_dict` and `enzyme_dict` later themselves.)

Reply:
In fact, I want to change `protein_dict` to `gene_product_dict` etc. this is more in keeping with how we named everything else in the pr